src/context.rs

```diff
@@ -888,13 +905,17 @@ impl Context {
     ///
     /// # Errors
     /// * if the hash list provided is too short or too long, a `WrongParamSize` wrapper error will be returned
-    pub fn policy_or(&mut self, policy_session: ESYS_TR, digest_list: DigestList) -> Result<()> {
+    pub fn policy_or(
+        &mut self,
+        policy_session_handle: SessionHandle,
```

Comment:
Can the session type here and for the other policy methods be restricted to `PolicySession`?

Reply:
Nope. It cannot be restricted to policy session. Because when you want to create a policy you needs to use a trial session in order to calculate things. Then you use a policy session when you want to use policy as authorization method.

Reply:
Why not use `Session` and verify below if it's either `Trial` or `Policy`?

Reply:
Because I was a little unsure if it was possible to pass something else besides a trial or a policy session to these functions. But if those two are the only options then it would make more sense to take a session and throw an error if it is not the correct type.

Reply:
Because I was a little unsure if it was possible to pass something else besides a trial or a policy session to these functions. But if those two are the only options then it would make more sense to take a session and throw an error if it is not the correct type.

Reply:
The specification is not to clear on the matter:
TCG_TPM2_r1p59_Part3_commands
> TPMI_SH_POLICY, policySession, handle for the policy session being extended

Reply:
Hrmm I have been convinced. They should all be session objects not handles.

Reply:
What was the resolution, then? The last comment says `Session` but the code is still `SessionHandle`?